### PR TITLE
[WIP] Misc Xenoarch Tweaks 2: Electric Boogaloo

### PR DESCRIPTION
**Pull request by BobdaBiscuit**
In which I go bughunting, add in a couple new features, and do some silly backend changes.
Nothing here yet, I'm just gonna do it bit by bit as usual.

**Shit to do**
- [ ] More robust handling of invalid trigger/effect combos
- [ ] Integrate new triggers into one time effects
- [ ] QoL shit
- [ ] Bugfix

**How to do it**
- [ ] Per Effect/Trigger blacklists and whitelists
- [ ] Overrides for invalid triggers
- [ ] Admin spawned artifacts have the validation system applied to it
- [x] Changed the isolation period from 20 seconds after activation to 20 seconds after deactivation
- [x] Coin on a string can now be used on coin triggers
- [x] Coin trigger rejects unknown coins
- [x] Coin trigger toggles added and prices adjusted
- [x] Electricity trigger simplified so it does not apply load, shit was too buggy
~~- [ ] Xenoguns having no icon~~
- [x] Field gen can read IDs inside a pda
~~- [ ] Alt click ejects beaker, ctrl click/kick starts scan on FTS/HI~~
- [ ] Prevent two identical pay2use triggers existing on one artifact.
- [x] Add in a Triggered() override that ignores the secondary effect 25% probability
- [ ] Add in logging for the user of speed/visible triggers
- [ ] Investigate/Fix effects happening through multiple z-levels

### CODE REVIEW

**Comment by ghost** [comment]
10 commits and 0 lines changed. off to a splendid start

**Comment by BobdaBiscuit** [comment]
Hue, I like to have my checklist here so I can track what I'm doing.

I also have no idea how to update my bleeding edge repo without causing a commit, so rip.

File: code/modules/research/xenoarchaeology/artifact/triggers/unknown_trigger_pay2use.dm [hunk 1]
```diff
@@ -80,25 +97,25 @@
...
 				else if(istype(inserted_coin, /obj/item/weapon/coin/plasma))
-					time_left += 45
+					time_left += coin_value * 4
 				else if(istype(inserted_coin, /obj/item/weapon/coin/uranium))
-					time_left += 50
+					time_left += coin_value * 6
 				else if(istype(inserted_coin, /obj/item/weapon/coin/diamond))
-					time_left += 100
+					time_left += coin_value * 10
 				else if(istype(inserted_coin, /obj/item/weapon/coin/phazon))
```
Comment (jknpj) [inline code/modules/research/xenoarchaeology/artifact/triggers/unknown_trigger_pay2use.dm]: could sum these rare coins in a single if(istype1 || istype2 || istype3)

Comment (BobdaBiscuit) [reply code/modules/research/xenoarchaeology/artifact/triggers/unknown_trigger_pay2use.dm]: Oh ye, don't mind me being a tard.

**Comment by Intigracy** [comment]
finish it my man

**Pull request closed without merging**